vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBoxRenderManager.java

```diff
@@ -80,5 +80,8 @@ private void render() {
         renderingRegistrations.add(rendering.getRegistration());
 
         comboBox.getDataController().reset();
+        if (comboBox.getValue() != null) {
+            comboBox.refreshValue();
+        }
```

Comment:
From what I understand the change makes sense, setting a renderer changes the data generator, so you need to run `refreshValue` again which relies on the data generator. 

Since this class is about managing the renderer, I would consider moving this code outside of this class to the call location that sets the renderer (`ComboBoxBase.setRenderer`). A code comment also can't hurt to explain why we need to regenerate the value.

Reply:
I get your point about moving this code out of that place. But if I put it to the `ComboBoxBase.setRenderer`, then it's code would have to like this:
```
    public void setRenderer(Renderer<TItem> renderer) {
        Objects.requireNonNull(renderer, "The renderer must not be null");

        renderManager.setRenderer(renderer);

        // renderer might add new dataGenerator -> refresh getValue() data
        // so it also contains the new dataGenerator data
        runBeforeClientResponse(ui -> {
            if (getValue() != null) {
                refreshValue();
            }
        });
    }
 ```
 
 that also does not seem right: a lot of details of `setRenderer` is leaked in it - that it might (or might not) set dataGenerator and that it's doe before client response...

Reply:
Makes sense, in that case I prefer the original approach, no need to overcomplicate it at this point 🙂. Just adding the comment there would be fine.

Reply:
Ok, I've just added a comment